portfolio/src/main/webapp/script.js

```diff
@@ -61,36 +58,36 @@ function createMessageElement(message) {
   timeElement.innerText = formatTimestamp(message.timestamp);
 
   const nameElement = document.createElement('span');
-  nameElement.setAttribute('class', 'name');
-  nameElement.innerText = message.name;
+  nameElement.setAttribute('class', 'nickname');
+  nameElement.innerText = message.nickname;
 
   const textElement = document.createElement('span');
   textElement.setAttribute('class', 'text');
   textElement.innerText = message.text;
-
-  const deleteButtonElement = document.createElement('button');
-  deleteButtonElement.setAttribute('class', 'delete');
-  deleteButtonElement.innerHTML="<img src=\"images/trash.svg\">";
-  deleteButtonElement.addEventListener('click', () => {
-    deleteMesage(message);
-
-    // Remove the message from the DOM.
-    messageElement.remove();
+  
+  fetch('/user-info').then(response => response.json()).then((userInfo) => {
```

Comment:
You fetch the user-info for each message separately. This is very expensive if you have many messages.
Better fetch it only once and maybe add a boolean param to this method whether to add the delete button.